components/infra-proxy-service/server/servers.go

```diff
@@ -63,6 +64,45 @@ func (s *Server) CreateServer(ctx context.Context, req *request.CreateServer) (*
 	}, nil
 }
 
+// ValidateWebuiKey validate the webui key
+func (s *Server) ValidateWebuiKey(ctx context.Context, req *request.ValidateWebuiKey) (*response.ValidateWebuiKey, error) {
+
+	if req.IsWebuiKey && req.Id != "" {
+		server, err := s.service.Storage.GetServer(ctx, req.Id)
+		if err != nil {
+			return nil, service.ParseStorageError(err, *req, "server")
+		}
+
+		// Get web ui key from secrets service
+		secret, err := s.service.Secrets.Read(ctx, &secrets.Id{Id: server.CredentialID})
+		if err != nil {
+			return nil, err
+		}
+
+		req.WebuiKey = GetWebuiKeyFrom(secret)
```

Comment:
Never modify request object.
Declare a local variable in the function to store the passed the WebUIKey and change it